test/sets.jl

```diff
@@ -89,10 +89,23 @@ s = Set([1])
 @test isequal(empty!(s), Set())
 
 # rehash!
-s = Set(1:5)
-k = s.dict.keys
-Base.rehash!(s)
-@test !isequal(s.dict.keys, k)
```

Comment:
Seems like we could do:
```
@test pointer(s.dict.keys) != pointer(k)
```
We need a `!=` test of some sort to check that the rehash was not a no-op (which would pass the new test).